internal/graph/check.go

```diff
@@ -143,19 +143,10 @@ func WithMaxConcurrentReads(limit uint32) LocalCheckerOption {
 	}
 }
 
-func WithDelegate(delegate CheckResolver) LocalCheckerOption {
-	return func(d *LocalChecker) {
-		d.delegate = delegate
-	}
-}
-
 func WithCachedResolver(opts ...CachedCheckResolverOpt) LocalCheckerOption {
 	return func(d *LocalChecker) {
-		cachedCheckResolver := NewCachedCheckResolver(
-			d,
-			opts...,
-		)
-		d.SetDelegate(cachedCheckResolver)
+		cachedCheckResolver := NewCachedCheckResolver(d, opts...)
```

Comment:
I've just noticed how odd this looks. `LocalChecker` can have a pointer to `CachedChecker`.

But `CachedChecker` _must_ receive a Checker, too.

https://github.com/openfga/openfga/blob/9e1f5366ca9ba672f11528c016f3e60a3b997a45/internal/graph/cached_resolver.go#L112-L123

So, these two structs depend on each other.

And then in this test `TestCachedCheckDatastoreQueryCount` we actually never call `NewCachedCheckResolver` 😆  and make assertions on `LocalChecker`.

https://github.com/openfga/openfga/blob/aa0c8162003a389263ee27c6be6731773bda0147/internal/graph/cached_resolver_test.go#L504-L506